### PR TITLE
[W5.4][W09-A2]Sun Shengran

### PR DESCRIPTION
**Pull request by dalessr**
Add an update function to AddressBook application

### CODE REVIEW

**Comment by jeffreygohkw** [comment]
Is it possible to update a person such that we can remove a certain field entirely? For example, if I want to remove a phone number from Adam using this update command.

**Comment by dalessr** [comment]
So far for the update command, users need to type every information about a person. Removing the phone number is invalid. I will make some improvements on this part.

**Comment by dalessr** [comment]
Thanks for your suggestion!

File: src/seedu/addressbook/commands/UpdateCommand.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package seedu.addressbook.commands;
```
Comment (nbriannl) [inline src/seedu/addressbook/commands/UpdateCommand.java]: Simple but elegant implementation of a new command

File: doc/UserGuide.md [hunk 2]
```diff
@@ -49,6 +49,22 @@ Examples:
 * `add John Doe p/98765432 e/johnd@gmail.com a/John street, block 123, #01-01`
 * `add Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/friend`
 
+### Updating a person: `update`
```
Comment (nbriannl) [inline doc/UserGuide.md]: Clear User Guide information

File: src/seedu/addressbook/parser/Parser.java [hunk 3]
```diff
@@ -160,6 +154,39 @@ private static boolean isPrivatePrefixPresent(String matchedPrefix) {
...
+                    matcher.group("email"),
+                    isPrivatePrefixPresent(matcher.group("isEmailPrivate")),
+
+                    matcher.group("address"),
+                    isPrivatePrefixPresent(matcher.group("isAddressPrivate")),
+
+                    getTagsFromArgs(matcher.group("tagArguments"))
+            );
+        } catch (IllegalValueException ive) {
+            return new IncorrectCommand(ive.getMessage());
```
Comment (nbriannl) [inline src/seedu/addressbook/parser/Parser.java]: I like the exception handling for incorrect command

File: doc/UserGuide.md [hunk 4]
```diff
@@ -49,6 +49,22 @@ Examples:
...
+> 
+> Put a `p` before the phone / email / address prefixes to mark it as `private`. `private` details can only
+> be seen using the `viewall` command.
+> 
+> Persons can have any number of tags (including 0)
+
+Examples: 
+* `update John Doe p/12345678 e/johnd@gmail.com a/John street, block 123, #01-01`
+* `update Betsy Crowe pp/1234567 e/betsycrowe@gmail.com pa/Newgate Prison t/criminal t/classmate`
+
```
Comment (damithc) [inline doc/UserGuide.md]: Um, how to specify which person to update? If you use the person name to identify which person to update, how do we update the name of a person?

File: src/seedu/addressbook/data/AddressBook.java [hunk 5]
```diff
@@ -84,6 +84,32 @@ public void addPerson(Person toAdd) throws DuplicatePersonException {
     }
 
     /**
+     * Updates a person to the address book.
+     *
+     * @throws PersonNotFoundException if the expected person does not exist.
+     */
+    public void updatePerson(Person toUpdate) throws PersonNotFoundException {
+
+        boolean checkPersonExist = false;
```
Comment (damithc) [inline src/seedu/addressbook/data/AddressBook.java]: Coding standard violation:exclamation:  variable name doesn't follow naming convention

File: src/seedu/addressbook/data/AddressBook.java [hunk 6]
```diff
@@ -84,6 +84,32 @@ public void addPerson(Person toAdd) throws DuplicatePersonException {
...
+
+        for (Person p : allPersons) {
+            if (p.getName().equals(toUpdate.getName())) {
+                p.setPhone(toUpdate.getPhone());
+                p.setEmail(toUpdate.getEmail());
+                p.setAddress(toUpdate.getAddress());
+                p.setTags(toUpdate.getTags());
+                checkPersonExist = true;
+            }
+        }
```
Comment (damithc) [inline src/seedu/addressbook/data/AddressBook.java]: This code can benefit from this guideline: [make the happy path prominent](https://nus-cs2103.github.io/website/book/codeQuality/practices/makeHappyPathProminent/)

File: src/seedu/addressbook/data/person/Person.java [hunk 7]
```diff
@@ -59,6 +59,12 @@ public UniqueTagList getTags() {
         return new UniqueTagList(tags);
     }
 
+    public void setPhone(Phone phone) { this.phone = phone; }
+
+    public void setEmail(Email email) { this.email = email; }
+
+    public void setAddress(Address address) { this.address = address; }
+
```
Comment (damithc) [inline src/seedu/addressbook/data/person/Person.java]: Coding standard violation:exclamation: layout/brace style problem.

**Pull request closed without merging**